spec/core/data_structures.md

```diff
@@ -4,195 +4,239 @@ Here we describe the data structures in the Tendermint blockchain and the rules
+| NextValidatorHash | slice of bytes (`[]byte`)     | MerkleRoot of the next validator set. The validators are first sorted by voting power (descending), then by address (ascending) prior to computing the MerkleRoot.                                                                                                                                                                                                                    | Must  be of length 32                                                                                                                                                                                                  |
+| ConsensusHash     | slice of bytes (`[]byte`)     | Hash of the protobuf encoded consensus parameters.                                                                                                                                                                                                                                                                                                                               | Must  be of length 32                                                                                                                                                                                                  |
+| AppHash           | slice of bytes (`[]byte`)     | Arbitrary byte array returned by the application after executing and commiting the previous block. It serves as the basis for validating any merkle proofs that comes from the ABCI application and represents the state of the actual application rather than the state of the blockchain itself. The first block's `block.Header.AppHash` is given by `ResponseInitChain.app_hash`. | This hash is determined by the application, Tendermint can not perform validation on it.                                                                                                                              |
+| LastResultHash    | slice of bytes (`[]byte`)     | `LastResultsHash` is the root hash of a Merkle tree built from `ResponseDeliverTx` responses (`Log`,`Info`, `Codespace` and `Events` fields are ignored). The first block has `block.Header.ResultsHash == MerkleRoot(nil)`, i.e. the hash of an empty input, for RFC-6962 conformance.                                                                                               | Must  be of length 32                                                                                                                                                                                                  |
```

Comment:
compared to the other fields, the description reads much more technical and detailed, and less insightful at the same time.